src/main/java/seedu/address/model/Model.java

```diff
@@ -19,14 +20,27 @@
     void resetData(ReadOnlyTaskBook newData);
 
     /** Returns the TaskBook */
-    ReadOnlyTaskBook getAddressBook();
+    ReadOnlyTaskBook getTaskBook();
 
     /** Deletes the given task. */
     void deleteTask(Task target) throws TaskNotFoundException;
 
     /** Adds the given task */
     void addTask(Task task);
-
+    
+    //undo and redo
+    /** Undo prev action that modifies data**/
+    Command undo();
+    
+    /** reset stack of redoable actions when a non undo modifying data command is called**/
+    void resetRedoables();
```

Comment:
Hmm, this seems to break encapsulation by exposing implementation details... I would expect that `recordStateBeforeChange` would automatically reset the redo stack.